presto-native-execution/testcontainers/README.md

```diff
@@ -0,0 +1,50 @@
+# Prestissimo: Functional Testing Using Containers
+
+This Java test framework allows you to run test cases by deploying Presto coordinator and worker nodes in containers.
+#### To set up Docker services and basic build tools on Ubuntu 22.04 x86_64 machine, run the following command:
+```
+apt install podman-docker
+apt install make
+apt install cmake
+```
+Note:  Run the following commands to fix an open bug related to CNI network in podman.
+Reference: https://bugs.launchpad.net/ubuntu/+source/libpod/+bug/2024394
+```
+curl -O http://archive.ubuntu.com/ubuntu/pool/universe/g/golang-github-containernetworking-plugins/containernetworking-plugins_1.1.1+ds1-3build1_amd64.deb
+dpkg -i containernetworking-plugins_1.1.1+ds1-3build1_amd64.deb
+```
+
+## Quick Start
+
+### 1. Build Presto using Maven
+The container images required for functional tests are integrated into the presto-native-execution and are built under a Maven profile named _docker-build_.
+
+```bash
+./mvnw clean install -DskipTests -Pdocker-build
+```
+
+### 2. Run functional tests
+#### Using Command Line
+Export the following environment variables:
+```bash
+export TESTCONTAINERS_RYUK_DISABLED=true
+export DOCKER_HOST=unix:///run/podman/podman.sock
+```
+Then, run the functional test using a command similar to this example for `TestPrestoContainerBasicQueries`:
+```bash
+./mvnw test -pl presto-native-execution -Dtest=com.facebook.presto.nativeworker.TestPrestoContainerBasicQueries
+```
+
+#### Using IntelliJ
```

Comment:
Seems like these tests require that both co-ordinator and worker are started with containers. How do we use this setup for debugging any worker issues ? With local runs we can suspend the test with a breakpoint and attach the debugger to the local presto_server. How do we do such debugging in this setup ?

Reply:
The goal for these tests would be to replace Tempto.  I think it would be possible to set up a remote debugger for both the C++ worker and the Java coordinator, but it's not meant to replace in-process tests.